test/functional/test_framework/wallet.py

```diff
@@ -370,6 +371,10 @@ def create_self_transfer(
             vsize = Decimal(168)  # P2PK (73 bytes scriptSig + 35 bytes scriptPubKey + 60 bytes other)
         else:
             assert False
+        if target_weight and not fee:  # respect fee_rate if target weight is passed
+            # the actual weight might be off by 3 WUs, so calculate based on that (see self._bulk_tx)
+            max_actual_weight = target_weight + 3
+            fee = get_fee((max_actual_weight + 3) // 4, fee_rate)
```

Comment:
Why are you incrementing by 3 twice
Please use `WITNESS_SCALE_FACTOR` const instead of just 4.

 Magic numbers makes my head hurts 😃 
 
 Should be worth it to define 3 in wallet as `PADDING_OFFSET` ?